src/platform/android/DnssdImpl.cpp

```diff
@@ -183,6 +183,20 @@ CHIP_ERROR ChipDnssdStopBrowse(intptr_t browseIdentifier)
     return CHIP_ERROR_NOT_IMPLEMENTED;
 }
 
+CHIP_ERROR extractProtocol(const char * serviceType, char * outServiceName, DnssdServiceProtocol & outProtocol)
+{
+    std::string serviceTypeStr(serviceType);
+    size_t index = serviceTypeStr.find(".");
+
+    ReturnErrorCodeIf(index == std::string::npos, CHIP_ERROR_INVALID_ARGUMENT);
+
+    CopyString(outServiceName, index + 1, serviceType);
+    outProtocol = (serviceTypeStr.substr(index + 1).compare(kOperationalProtocol) == 0) ? DnssdServiceProtocol::kDnssdProtocolTcp
+                                                                                        : DnssdServiceProtocol::kDnssdProtocolUdp;
```

Comment:
Why is there a desire to extract teh TCP versus UDP?

_matter._tcp is a for protocol than can run over UDP, and in the future, also TCP. Same for _matterc._udp. The `._udp`/`._tcp` are not fundamental. Therefor is there a reason why mProtocol is used/added?

The API allows it, but what is the end-usage of it? In other words, why this cahnge now? If it steers to TCP/UDP usage elsewhere, there will be different problems.

Reply:
This code shows '.' The purpose is to change the following "_tcp" String to DnssdServiceProtocol::kDnssdProtocolTcp, and to replace the "_udp" String to DnssdServiceProtocol::kDnssdProtocolUdp.

Example : 
1. If "_matter._tcp" value is entered in Java Layer, "_matter" is stored in mType variable and DnssdServiceProtocol::kDnssdProtocolTcp value is stored in mProtocol variable.
2. If "_matterc._udp" value is entered in Java Layer, "_matterc" is stored in mType variable and DnssdServiceProtocol::kDnssdProtocolUdp value is stored in mProtocol variable.

kOperationalProtocol has "_tcp" value stored.

Please check if the part you gave us your opinion is about the possibility that other values other than "_tcp" and "_udp" will be delivered.

Reply:
The question is about "why is this needed now". I am trying to understand if the fact that this feature was not implemented broke anything, because it should not have broken anything.

Reply:
I'm sorry. I didn't understand you correctly.

I'm going to give you my opinion on that

1. For Linux chip-tool, a similar code has already been added below.
https://github.com/project-chip/connectedhomeip/blob/733afaabd512cbf396d3edf09805d1054a6ec5f8/src/platform/Linux/DnssdImpl.cpp#L631

2. I don't know the exact reproduction path, but in order to clarify this part, I changed the code to the following location and tested it to see if it worked. (Initial value specified in Tcp)
https://github.com/project-chip/connectedhomeip/blob/733afaabd512cbf396d3edf09805d1054a6ec5f8/src/lib/dnssd/platform/Dnssd.h#L69

-> DnssdServiceProtocol mProtocol = DnssdServiceProtocol::DnssdServiceProtocol::kDnssdProtocolTcp;

Judging from this, it can be seen that there is a possibility of an issue, not just reproduced. I think it will be initialized to zero due to the code below, but I think this code should be added because there is a history of issues.

https://github.com/project-chip/connectedhomeip/blob/733afaabd512cbf396d3edf09805d1054a6ec5f8/src/platform/android/DnssdImpl.cpp#L316

Therefore, I think you should set the mProtocol variable to clarify this part.